ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/TableExport.java

```diff
@@ -232,16 +247,16 @@ private Path exportDir(Path exportDir) throws SemanticException {
       }
     }
 
-    private Path metaDataExportFile() {
-      return new Path(exportRootDir, EximUtil.METADATA_NAME);
+    private Path metaDataExportFile() throws SemanticException {
+      return new Path(exportRootDir(), EximUtil.METADATA_NAME);
```

Comment:
not sure again. what do you mean not create directory and use variable. the reason being the directory has to be created for replication, however there are certain decisions we take based on table type etc which is outside the scope of this class and hence we do lazy creation of the directory only when the methods are called and not otherwise.